Update Macintosh package libraries

milljm opened the pull request:
Update all packages involved with the previous inability to run Peacock.

Closes #20127

Commit:
Update Macintosh package libraries

Update all packages involved with the previous inability to run Peacock.

Closes #20127

milljm commented on conda/mpich/conda_build_config.yaml:
```diff
@@ -203,11 +225,11 @@ moose_sysroot:                                              # [linux]
 
 moose_libclang:                                             # [osx]
-  - libclang 12.0.1 default_he082bbe_4                      # [not arm64 and osx]
+  - libclang 13.0.0 default_he082bbe_0                      # [not arm64 and osx]
```
I know this is strange, but this is what pyqt/qt _demands_. And it seems to not conflict with our 12.x requirements.

cticenhour replied:
I seem to remember always having to update `libclang` with the main clang version we use in the package. I find it interesting that you didn't need to adjust `libcxx` as well.

milljm commented on conda/libmesh-vtk/build.sh:
```diff
@@ -30,7 +30,8 @@ cmake .. -G "Ninja" \
 
-ninja install -v
+CORES=$(echo "${CPU_COUNT:-2} / 2" | bc)
+ninja install -v -j $CORES
```
I found while building on my machine, it is destroyed by oversubscribing. I think Conda treats threads as cores.

cticenhour replied:
In favor of this, as conda build has definitely used more than I thought it should in the past. 👍🏻

cticenhour commented:
I'm seeing several instances in `moose-tools` where we have earlier versions (say `vtk`, `numpy`, etc.) for Python 3.6 and 3.7 versus the restored versions (3.8 and 3.9). Any chance these can be updated so that we're using consistent versions?

milljm commented:
> I'm seeing several instances in `moose-tools` where we have earlier versions for certain packages (say `vtk`, `numpy`, etc.) for Python 3.6 and 3.7 versus those packages corresponding to 3.8 and 3.9. Any chance these can be updated so that we're using consistent versions?

Indeed, working on those now. I see an updated PyQt for Python 3.7. So that is a _good_ sign!

cticenhour commented:
Oops, I jumped the gun on my comment - good to hear!

Commit:
Changes to update Python 3.7 and 3.6

Refs #20127

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/941007/) on 0dddb4b wanted to post the following:

View the site [here](https://mooseframework.inl.gov/docs/PRs/20128/site)

This comment will be updated on new commits.

moosebuild commented:
Job [Coverage](https://civet.inl.gov/job/941017/) on 0dddb4b wanted to post the following:


## Framework coverage

Coverage did not change


## Modules coverage

Coverage did not change

## Full coverage reports

<details>
<summary>Reports</summary>

- [`framework`](https://mooseframework.inl.gov/docs/PRs/20128/coverage/framework)
- [`chemical_reactions`](https://mooseframework.inl.gov/docs/PRs/20128/coverage/chemical_reactions)
- [`combined`](https://mooseframework.inl.gov/docs/PRs/20128/coverage/combined)
- [`contact`](https://mooseframework.inl.gov/docs/PRs/20128/coverage/contact)
- [`external_petsc_solver`](https://mooseframework.inl.gov/docs/PRs/20128/coverage/external_petsc_solver)
- [`fluid_properties`](https://mooseframework.inl.gov/docs/PRs/20128/coverage/fluid_properties)
- [`fsi`](https://mooseframework.inl.gov/docs/PRs/20128/coverage/fsi)
- [`functional_expansion_tools`](https://mooseframework.inl.gov/docs/PRs/20128/coverage/functional_expansion_tools)
- [`geochemistry`](https://mooseframework.inl.gov/docs/PRs/20128/coverage/geochemistry)
- [`heat_conduction`](https://mooseframework.inl.gov/docs/PRs/20128/coverage/heat_conduction)
- [`level_set`](https://mooseframework.inl.gov/docs/PRs/20128/coverage/level_set)
- [`misc`](https://mooseframework.inl.gov/docs/PRs/20128/coverage/misc)
- [`navier_stokes`](https://mooseframework.inl.gov/docs/PRs/20128/coverage/navier_stokes)
- [`peridynamics`](https://mooseframework.inl.gov/docs/PRs/20128/coverage/peridynamics)
- [`phase_field`](https://mooseframework.inl.gov/docs/PRs/20128/coverage/phase_field)
- [`porous_flow`](https://mooseframework.inl.gov/docs/PRs/20128/coverage/porous_flow)
- [`ray_tracing`](https://mooseframework.inl.gov/docs/PRs/20128/coverage/ray_tracing)
- [`rdg`](https://mooseframework.inl.gov/docs/PRs/20128/coverage/rdg)
- [`reactor`](https://mooseframework.inl.gov/docs/PRs/20128/coverage/reactor)
- [`richards`](https://mooseframework.inl.gov/docs/PRs/20128/coverage/richards)
- [`stochastic_tools`](https://mooseframework.inl.gov/docs/PRs/20128/coverage/stochastic_tools)
- [`tensor_mechanics`](https://mooseframework.inl.gov/docs/PRs/20128/coverage/tensor_mechanics)
- [`xfem`](https://mooseframework.inl.gov/docs/PRs/20128/coverage/xfem)

</details>



This comment will be updated on new commits.

loganharbour commented on conda/mpich/conda_build_config.yaml:
```diff
@@ -2,7 +2,7 @@
 
 moose_libmesh:
-  - moose-libmesh 2022.01.19 build_1
+  - moose-libmesh 2021.12.15 build_2
```
```suggestion
  - moose-libmesh 2022.01.19 build_2
```

Commit:
Rebase off next, bump libmesh version

Refs #20127

milljm commented:
While this may pass... we're seeing an interesting occurrence of hung processes. We need to investigate further.

cticenhour commented:
Note to ourselves later - hanging test harness in modules combined runs seems to be related to python: 
```
/Users/user/miniconda3/envs/test_clang13/bin/python -c from multiprocessing.resource_tracker import main;main(6)
```
This system hang seems to occur only when large numbers of tests are being run at once (like "modules combined") instead of when modules or framework are tested individually.

Commit:
Pin a min/max to setuptools

Pin setuptools for those packages which use it. This should hopefully
prevent the strange python multithreading hangups we've been seeing.

Refs: #20127
